File: add_binary/add_binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        carry = 0
        s = []
        i = len(a) - 1
        j = len(b) - 1
        while i >= 0 or j >= 0 or carry:
            if i >= 0:
                carry += int(a[i])
                i -= 1
            if j >= 0:
                carry += int(b[j])
                j -= 1
            s.append(str(carry%2))
            carry //= 2
        return ''.join(reversed(s))
    

if __name__ == '__main__':
    solution = Solution()
    print(solution.addBinary('11', '1'))
    print(solution.addBinary('1010', '1011'))

File: add_binary/add_binary.ts
function addBinary(a: string, b: string): string {
  let ans: string = "";
  let carry: number = 0;
  
  let i: number = a.length - 1;
  let j: number = b.length - 1;
  
  while (i >= 0 || j >= 0 || carry > 0) {
    if (i >= 0) {
      carry += parseInt(a[i]);
      i--;
    }
    
    if (j >= 0) {
      carry += parseInt(b[j]);
      j--;
    }
    ans += (carry % 2).toString();
    carry = Math.floor(carry / 2);
    
  }
  return ans.split("").reverse().join("");
};

console.log(addBinary("11", "1"));
console.log(addBinary("1010", "1011"));